### PR TITLE
feat: implement edit highlight

### LINKED ISSUE #886: feat: Make highlights editable
As a user I should be able to edit a highlight. 

Screen Shot 2023-02-20 at 9 35 07 AM

https://www.figma.com/file/OpVX6WT7dmWqnwRuEvADMF/OpenSauced-%E2%80%94-Insights-(insights.opensauced.pizza)?node-id=5876%3A7583&t=KG4cKsqY6B6hvToP-0

### PR DESCRIPTION
**Pull request by OgDev-01**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description
This PR:

- creates a dialog leveraging https://www.radix-ui.com/docs/primitives/components/dialog
- implements update highlights functionality
- Adds validation for max character limiting
- Adds Ui for character limiting
- implements highlight edit guard 

## Related Tickets & Documents
Fixes #886 

## Mobile & Desktop Screenshots/Recordings

image

image

### Error state
image

### Character limit validation
image

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

File: components/molecules/ContributorHighlight/contributor-highlight-card.tsx [hunk 1]
```diff
@@ -40,98 +63,232 @@ const ContributorHighlightCard = ({ title, desc, prLink, user }: ContributorHigh
...
+
+    if (isValidGhUrl) {
+      const { apiPaths } = generateApiPrUrl(highlight.prLink);
+      const { repoName, orgName, issueId } = apiPaths;
+      setLoading(true);
+      const res = await fetchGithubPRInfo(orgName, repoName, issueId);
+
+      if (res.isError) {
+        setLoading(false);
+        setError("Please provide a valid github pull url");
```
Comment (bdougie) [inline components/molecules/ContributorHighlight/contributor-highlight-card.tsx:83]:
```suggestion
        setError("Please provide a valid github pull url");
```
```suggestion
        setError("Please provide a valid github pull request url");
```

File: components/molecules/ContributorHighlight/contributor-highlight-card.tsx [hunk 2]
```diff
@@ -40,98 +63,232 @@ const ContributorHighlightCard = ({ title, desc, prLink, user }: ContributorHigh
...
+          ToastTrigger({ message: "Highlights Updated Successfully", type: "success" });
+          mutate(`users/${user}/highlights`);
+          setOpen(false);
+        } else {
+          setLoading(false);
+          setError("An error occurred while updating!!!");
+        }
+      }
+    } else {
+      setError("Please provide a valid github pull url");
```
Comment (bdougie) [inline components/molecules/ContributorHighlight/contributor-highlight-card.tsx:101]:
```suggestion
      setError("Please provide a valid github pull request url");
```

**Review by bdougie — CHANGES_REQUESTED** [review]
needs to be "pull request"

**Review by bdougie — APPROVED** [review]
Looks good. I committed some copy updates

**Comment by shamimbinnur** [comment]
@OgDev-01  I am gonna replicate this [dialog.tsx](https://github.com/open-sauced/insights/blob/886-new-update-highlight/components/molecules/Dialog/dialog.tsx) on my branch and use it for the issue #853. To avoid conflict I need confirmation that this is reusable for future work and you are not gonna make any changes to the [dialog.tsx](https://github.com/open-sauced/insights/blob/886-new-update-highlight/components/molecules/Dialog/dialog.tsx) at least in this PR.

**Comment by bdougie** [comment]
merge the PR. You can work off beta to get the components final state @shamimbinnur

File: components/molecules/ContributorHighlight/contributor-highlight-card.tsx (hunk 1 above)
Comment (Charliebrickman75) [reply components/molecules/ContributorHighlight/contributor-highlight-card.tsx]: Did this work?

**Pull request merged**